### PR TITLE
[go] [generic] add pull request support without signing

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

1. Should we display some github log warning?
2. This sort of makes the check `IsPresubmitTest` not necessary. Is that OK? I'm not sure if we want to continue gating on that, or maybe we want to add an explicit flag to users like "snapshot" that is required to allow pull_requests to generate the predicate (that way users don't accidentally generate unsigned provenance)

### CODE REVIEW

File: internal/builders/go/pkg/provenance.go [hunk 1]
```diff
@@ -70,6 +70,11 @@ func GenerateProvenance(name, digest, command, envs, workingDir string, s signin
 		return nil, err
 	}
 
+	// If the environment does not have access to an OIDC provider, use a nil one.
+	if provider == nil && !github.HasOIDCClient() {
```
Comment (laurentsimon) [inline internal/builders/go/pkg/provenance.go:74]: are there cases where provider is not `nil` and has not OIDC client? Existing code suggest no.

File: internal/builders/generic/attest.go [hunk 2]
```diff
@@ -80,29 +85,19 @@ run in the context of a Github Actions workflow.`,
 			}
 			if provider != nil {
```
Comment (laurentsimon) [inline internal/builders/generic/attest.go:86]: Can we remove the condition `if provider != nil`?

**Review by laurentsimon — APPROVED** [review]
LGTM. I agree with you it'd be nice for users to specify an input acknowledging they want to opt-in for dry run. This will force them to read the doc that explains that on triggers with no OIDC clients, they should not expect it. Otherwise I suspect some user will miss it and complain?

if we don't want to commit to a new input argument, we could start by having this clearly in the documentation and re-visit in a few months.

**Review by ianlewis — APPROVED** [review]

**Comment by ianlewis** [comment]
Updates #131

**Comment by ianlewis** [comment]
Updates #358

**Comment by ianlewis** [comment]
@asraa This is pretty old. Should we close it?

**Comment by asraa** [comment]
> @asraa This is pretty old. Should we close it?

I think so. If we achieve this the better route to go is to refactor them to use the signing actions

**Pull request closed without merging**